translate/Main.hs

```diff
@@ -98,15 +98,21 @@ doTranslate :: MonadIO m => [DynSignal] -> Concept (State DynSignal) (Transition
 doTranslate signs circuit = do
     case validate signs circuit of
         Valid -> do
-            let initStrs = map (\s -> (show s, False)) signs {- TODO: don't hard-code to false -}
+            let initStrs = map (\s -> (show s, (getDefined $ initial circuit s))) signs
             let arcStrs = map (\(from, to) -> (show from, show to)) (arcs circuit)
             let inputSigns = filter ((==Input) . interface circuit) signs
             let outputSigns = filter ((==Output) . interface circuit) signs
             let internalSigns = filter ((==Internal) . interface circuit) signs
             liftIO $ putStr $ genSTG inputSigns outputSigns internalSigns initStrs arcStrs
             return ()
-        UnusedSignal ss -> liftIO $ putStr $ "Error. The following signals are not declared as input, output or internal: \n"
-            ++ unlines (map show ss)
+        Invalid ss is us -> liftIO $ do
```

Comment:
Let's use clearer names here: `ss`, `is` and `us` are too cryptic. Perhaps just reuse names from the validation function.